text/0617-cloudfront-oac-l2.md

```diff
@@ -0,0 +1,648 @@
+
+#### Using OAC for a SSE-KMS encrypted S3 origin
+
+If the objects in the S3 bucket origin are encrypted using server-side encryption with
+AWS Key Management Service (SSE-KMS), the OAC must have permission to use the AWS KMS key.
+A statement needs to be added to the KMS key policy to give the OAC permission to use the KMS key.
+
+```ts
+const myKmsKey = new kms.Key(this, 'myKMSKey');
+const myBucket = new s3.Bucket(this, 'mySSEKMSEncryptedBucket', {
+  encryption: s3.BucketEncryption.KMS,
+  encryptionKey: kmsKey,
+  objectOwnership: s3.ObjectOwnership.BUCKET_OWNER_ENFORCED,
+});
+new cloudfront.Distribution(this, 'myDist', {
+  defaultBehavior: { 
+    origin: new origins.S3Origin(myBucket) // Automatically creates an OAC
+  },
+});
+```
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `status/api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @colifran
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are launching a new L2 construct `OriginAccessControl` for CloudFront (`aws-cdk-lib/aws-cloudfront`). We are also launching some modifications to
+the existing`S3Origin` construct in the `aws-cdk-lib/aws-cloudfront-origins` module.
+
+### Why should I use this feature?
+
+With this new feature, you can follow AWS best practices of using IAM service principals to authenticate with your AWS origin. This ensures users only
+access the content in your AWS origin through your specified CloudFront distribution. OAC also supports new AWS regions launched after December 2022
+and S3 origins that use SSE-KMS encryption.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+This feature has been highly requested by the community since August 2022 when Origin Access Control was launched (195 upvotes on the
+[GitHub issue](https://github.com/aws/aws-cdk/issues/21771)). Although the L1 construct `CfnOriginAccessControl` exists, users currently need to remove
+the OAI automatically configured by the existing `S3Origin` construct which is a subpar user experience. We want to make it easier for users to follow
+AWS best practices and secure their CloudFront origins.
+
+### Why should we _not_ do this?
+
+Users who want to use OAC may have already found workarounds using the L1 construct.
+
+### What is the technical solution (design) of this feature?
+
+This feature will be introduced under a feature flag `@aws-cdk/aws-cloudfront:useOriginAccessControl` as the current default configuration
+for S3 origins using OAI is still supported.
+
+#### New `OriginAccessControl` L2 Construct
+
+```ts
+export interface IOriginAccessControl extends IResource {
+  /**
+   * The unique identifier of the origin access control.
+   * @attribute
+   */
+  readonly originAccessControlId: string;
+}
+
+abstract class OriginAccessControlBase extends Resource implements IOriginAccessControl {
```

Comment:
This base class is pretty thin. Any reason we couldn't just let `OriginAccessControl` extend `Resource` and implement `IOriginAccessControl` directly? If the class grows larger and needs a base class then we can always factor the logic out into a base class later.

Reply:
Agreed it's kind of unnecessary in this case. No good reason other than I was following the typical L2 construct structure which has a base class.